common/djangoapps/track/middleware.py

```diff
@@ -164,31 +163,54 @@ def enter_request_context(self, request):
         )
 
     def get_session_key(self, request):
-        """ Gets and encrypts the Django session key from the request or an empty string if it isn't found."""
+        """
+        Gets a key suitable for representing this Django session for tracking purposes.
+
+        Returns an empty string if there is no active session.
+        """
         try:
-            return self.encrypt_session_key(request.session.session_key)
+            return self.substitute_session_key(request.session.session_key)
         except AttributeError:
+            # NB: This can hide a missing SECRET_KEY
             return ''
 
-    def encrypt_session_key(self, session_key):
-        """Encrypts a Django session key to another 32-character hex value."""
+    def substitute_session_key(self, session_key):
+        """
+        Deterministically generate a tracking session key from the real one.
+
+        If a session key is not provided, returns empty string.
+
+        The tracking session ID is a 32-character hexadecimal string (matching
+        Django session key format for convenience, and in case something
+        downstream makes assumptions.) The tracking ID does not allow recovery
+        of the original session key but will always be the same unless server
+        secrets are changed, and will be unique for each session key.
```

Comment:
I think I might want to add a test showing that different session keys get different tracking IDs. :-)